aronnax/core.py

```diff
@@ -158,6 +171,18 @@ def v_point_variable_3d(grid, func):
             v_variable_3d[i,:,:] = f(X, Y)
     return v_variable_3d
 
+def time_series_variable(nTimeSteps, dt, func):
+    '''Input generator for a time series variable. If passed a function, then that function can depend on the number of timesteps, `nTimeSteps`, and the timestep, `dt`.'''
+
+    ts_variable = np.zeros((nTimeSteps))
+
+    for i, f in enumerate(func):
```

Comment:
Fair enough.

It was meant to check the element in the list, not the list itself - that was a mistake. I set it to take a list to maintain consistency with the other generator functions assuming that it would only ever get passed a list of length one. I think I'm not defensive enough in my programming. I've added an assertion that the list length is one and some tests to verify that it produces the expected output.

I will endeavour to add unit tests for new code from now on.